### PR TITLE
calypso-build: add webpack config for output.jsonpFunction

### PR DESCRIPTION
**Pull request by creativecoder**
#### Changes proposed in this Pull Request

* Add configuration for output.webpackjsonpFunction in the calypso-build package

The [webpack docs state](https://v4.webpack.js.org/configuration/output/#outputjsonpfunction)

> If multiple webpack runtimes (from different compilations) are used on the same webpage, there is a risk of conflicts of on-demand chunks in the global namespace.

This seems to be a source of WordPress plugin conflicts, when multiple plugins use webpack to build js. For example: https://github.com/Automattic/jetpack/issues/17289

Adding this config allows plugins that use calypso-build's webpack config, like Jetpack, to customize the webpack jsonp global variable name and avoid these conflicts.

Note that this config is changing to `chunkLoadingGlobal` in [webpack 5](https://webpack.js.org/configuration/output/#outputchunkloadingglobal), so we might want to address that in some way, as well.

#### Testing instructions

* Add `output-jsonp-function` to a webpack build process that uses `getWebpackConfig` from calypso-build (Jetpack example: https://github.com/Automattic/jetpack/pull/17312/commits/33815b44dd0d69f039aa543ef3dc71f37e338820)
* Check the output js, and see that `window["webpackJsonp"]` uses what you specified for the global variable

Required for fixing https://github.com/Automattic/jetpack/issues/17289

### CODE REVIEW

**Review by scinos — APPROVED** [review]

**Review by jsnajdr — APPROVED** [review]
Nice! :+1:

File: packages/calypso-build/webpack.config.js [hunk 1]
```diff
@@ -54,6 +54,7 @@ function getWebpackConfig(
 		'output-path': outputPath = path.join( process.cwd(), 'dist' ),
 		'output-filename': outputFilename = '[name].js',
 		'output-library-target': outputLibraryTarget = 'window',
+		'output-jsonp-function': outputJsonpFunction = 'webpackJsonp',
```
Comment (jsnajdr) [inline packages/calypso-build/webpack.config.js:57]: Maybe also add a line to the docblock comment few lines above? The existing documentation there is not terribly helpful though 🙁  (`output-path` sets output path and `output-library-target` sets output library target, who would have guessed that)

Comment (creativecoder) [reply packages/calypso-build/webpack.config.js:58]: 👍 Done in https://github.com/Automattic/wp-calypso/pull/46252/commits/b42eaaa94c94f7cf28832b4fccbf0f71875a002b

**Comment by creativecoder** [comment]
@scinos @jsnajdr Thanks for the prompt review!

How do you recommend getting the calypso-build package updated so this change can be used externally? Okay to change package version in this PR (to `6.2.1`) and update changelog.md, or should that be handled separately?

**Comment by jsnajdr** [comment]
@creativecoder We usually do the release chores (changelog, version bump) as a separate PR. There is a P2 post about how to publish packages to NPM (with a video!) here: p4TIVU-9mL-p2. The [NPM publishing docs in the Calypso repo](https://github.com/Automattic/wp-calypso/blob/master/docs/monorepo.md#publishing) also look reasonably up to date.

**Pull request merged**